File: API_SC_01_08Abril/Joao_Paulo/joao_paulo_lopes/q2_escada_transport.js
import { GetNumber, GetNumberAbove, GetNumberBetween, GetNumberInteger, GetNumberIntegerBetween, input,print } from "./utils.js"

function ConversorPesoCalorias(Peso, Sexo){//função que converte o peso(kg) em calorias

    if(Sexo === 1){//se for uma mulher
        return Peso * 9000
    }
    //caso seja um homem
    return Peso * 9400
}


function ComputarCaloriaTransport(Tempo,Sexo){//função que calcula quantas calorias a pessoa vai perder em 1 dia no transport
    let UnidadeMinutosGastos = Tempo / 6
    if(Sexo === 0){//caso seja homem
        UnidadeMinutosGastos = Tempo / 5
    }
    //caso seja mulher
    return 100 * UnidadeMinutosGastos

}


function ComputarCaloriaEscada(Tempo,Sexo){//função que calcula quantas calorias a pessoa vai perder em 1 dia na escada
    let UnidadeMinutosGastos = Tempo / 8
    if(Sexo === 0){//caso seja homem
        UnidadeMinutosGastos = Tempo / 7
    }
    //caso seja mulher
    return 100 * UnidadeMinutosGastos

}


function main(){

    //entrada de dados
    const PesoAtual = GetNumberAbove('Informe seu peso atual[Kg]: ',0)
    print('Masculino : [0] | Feminino : [1]')
    //transformando o sexo em numero para ficar mais facil de verificar
    const Sexo = GetNumberBetween('Informe seu sexo: ',0,1)
    const PerderPeso = GetNumberAbove('Informe quantos quilos[Kg] você quer perder: ',0)
    const DiasAtividadeFisica = GetNumberIntegerBetween('Informe quantos dias na semana você fará atividade física: ', 1,7)
    const TempoTreinoDia = GetNumberAbove('Informe quanto tempo você irá treinar no dia[Mins]: ',0)
    const PorcentagemTempoTransport = GetNumberIntegerBetween('Qual a proporção de tempo alocada para o Transport[%]: ',0,100)

    //processamento dos dados
    const PorcentagemTempoEscada = 100 -  PorcentagemTempoTransport
    const TempoMinutosTransport = TempoTreinoDia * (PorcentagemTempoTransport / 100)//total de minutos gastos no transport por dia
    const TempoMinutosEscada = TempoTreinoDia * (PorcentagemTempoEscada / 100) //total de minutos gasto na escada por dia
    const GastoCaloricoNecessario = ConversorPesoCalorias(PerderPeso,Sexo)
    //gasto calorico em um dia e em uma semana
    const GastoCaloricoTrasport = ComputarCaloriaTransport(TempoMinutosTransport,Sexo)
    const GastoCaloricoEscada = ComputarCaloriaEscada(TempoMinutosEscada,Sexo)
    const GastoCaloricoSemanal = (GastoCaloricoEscada + GastoCaloricoTrasport) * DiasAtividadeFisica 
    //para calcular em quantas semanas ele vai perder esse peso
    let SomaGastoCalorico = 0
    let ContadorSemanas = 0

    while(GastoCaloricoNecessario > SomaGastoCalorico){
        ContadorSemanas++
        SomaGastoCalorico += GastoCaloricoSemanal
    }

    //exibição dos dados
    const Tabela = `
    VOCÊ ALCANÇARÁ SEU OBJETIVO EM                : [${ContadorSemanas} SEMANA(S)]
    QUANTOS MINUTOS DE ESCADA VOCÊ DEVE FAZER     : [${TempoMinutosEscada} MINUTOS](DIÁRIOS)
    QUANTOS MINUTOS DE TRANSPORT VOCÊ DEVE FAZER  : [${TempoMinutosTransport} MINUTOS](DIÁRIOS)`

    print(Tabela)
     
}


main()

File: API_SC_01_08Abril/pedro guilherme leal de araujo/q2_escada_transport.py
import math

def questions():
    sexo = input("Qual seu sexo? ").lower()
    while(sexo!= "feminino"):
        print("valor invalido (feminino ou masculino)")
        sexo = input("Qual seu sexo? ").lower()
    atual_peso = float(input("Qual seu peso atual? "))
    perda_peso = float(input("Quantos quilos você quer perder? "))
    tempo_de_treino = int(input("Quantos minutos você irá treinar por dia? "))
    dias_treino = int(input("Quantos dias você irá treinar por semana? "))
    while(dias_treino>7):
        print("valor invalido")
        dias_treino = int(input("Quantos dias você irá treinar por semana? "))
    tempo_p_transpir = float(input("Qual a proporção em '%' do seu treino de transport? "))
    while(tempo_p_transpir>100):
        print("valor invalido")
        tempo_p_transpir = float(input("Qual a proporção em '%' do seu treino de transport? "))
    tempo_p_transpir = tempo_de_treino*(tempo_p_transpir/100)
    tempo_p_escad =tempo_de_treino-tempo_p_transpir



    return sexo,atual_peso,perda_peso,tempo_de_treino,dias_treino,tempo_p_transpir,tempo_p_escad


def calculo(sexo,atual_peso,perda_peso,tempo_de_treino,dias_treino,tempo_p_transpir,tempo_p_escad):
    

    calorias_correspondentes = perda_peso*7000
    if(sexo=="feminino"):
        gasto_semana_transport = math.floor(dias_treino*(tempo_p_transpir/6)*100)
        gasto_semana_escada = math.floor(dias_treino*(tempo_p_escad/8)*100)
    elif(sexo=="masculino"):
        gasto_semana_transport = math.floor(dias_treino*(tempo_p_transpir/5)*100)
        gasto_semana_escada = math.floor(dias_treino*(tempo_p_escad/7)*100)
    
    gasto_total = gasto_semana_escada+gasto_semana_transport

    print(f"Você irá precisar de aproximadamente {math.floor(calorias_correspondentes/gasto_total)} semanas para atingir seu objetivo!!")
    #print(calorias_correspondentes/gasto_semana_transport)
    print(f"Você deve fazer {tempo_p_escad} minutos de escada por dia")
    print(f"Você deve fazer {tempo_p_transpir} minutos de transport por dia")
 

sexo,atual_peso,perda_peso,tempo_de_treino,dias_treino,tempo_p_transpir,tempo_p_escad = questions()
calculo(sexo,atual_peso,perda_peso,tempo_de_treino,dias_treino,tempo_p_transpir,tempo_p_escad)
    
    